### PR TITLE
Add multi-select component using multiselect-react-dropdown

### PR DESCRIPTION
**Pull request by RyanPrice-King**

### CODE REVIEW

File: src/components/formRow/formRow.comp.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React, { useState } from 'react';
 import { orderBy } from 'natural-orderby';
 import { toast } from 'react-toastify';
+import { Multiselect } from 'multiselect-react-dropdown';
```
Comment (dsternlicht) [inline src/components/formRow/formRow.comp.tsx:4]: Why not using a more popular (and maintained) library such as `react-select`?
https://react-select.com/home

File: src/components/formRow/formRow.comp.tsx [hunk 2]
```diff
@@ -172,19 +174,34 @@ export const FormRow = withAppContext(({ context, field, direction, showReset, o
...
+            let selection = [];
+            var parsedJson = JSON.parse(JSON.stringify(selectedList))
+            for(var i = 0; i < parsedJson.length; i++) {
+              selection.push(JSON.stringify(parsedJson[i].value));
+            };
+            changeCallback(field.name, selection.toString().replace(/"/g, ''))
+          };
+
           return (
-            <select {...inputProps()}>
```
Comment (dsternlicht) [inline src/components/formRow/formRow.comp.tsx:176]: For non multi select, maybe we should use a simple selectbox?

**Review by dsternlicht — CHANGES_REQUESTED** [review]

File: src/components/formRow/formRow.comp.tsx (hunk 1 above)
Comment (RyanPrice-King) [reply src/components/formRow/formRow.comp.tsx:4]: I had problems with others importing the `finalOptions` object and found this one to contain additional parameters that were useful like the `selectionLimit` for multi.  Also, it allowed me to use the same component for single and multi-select.

File: src/components/formRow/formRow.comp.tsx (hunk 2 above)
Comment (RyanPrice-King) [reply src/components/formRow/formRow.comp.tsx:176]: I did originally, but seemed redundant to me as I could use the same component for both where multiple lines were originally duplicated across both components and this reduced the lines of code needed and simplified the code.  Also, with this approach I felt it is better for the end-user to not have to think about other types and to just add a simple multi switch to the select type.

**Comment by dsternlicht** [comment]
@RyanPrice-King this is how it displays right now after I pulled the changes:
Screen Shot 2021-12-14 at 8 38 27
Screen Shot 2021-12-14 at 8 38 34

**Review by dsternlicht — CHANGES_REQUESTED** [review]
Something is wrong with the styles:
https://github.com/dsternlicht/RESTool/pull/193#issuecomment-993207308

**Comment by RyanPrice-King** [comment]
image
It works with optionSource.

I will need to find out why it is not working with config options I think.

**Comment by dsternlicht** [comment]
> image It works with optionSource.
> 
> I will need to find out why it is not working with config options I think.

Ok. For non-multi source select boxes, let's simply use the regular selectbox (as we had until now).

**Comment by RyanPrice-King** [comment]
Ok, I have now found the problem and fixed it and added an object check like what you had before as that was one of the issues here, but if you prefer to split it still I will do that as well. and push the changes.

**Comment by RyanPrice-King** [comment]
Either way, can I ask you not to merge yet please as I am testing for other improvements as I build the site I require and may make further improvements.

**Comment by dsternlicht** [comment]
@RyanPrice-King no problem, I'm not merging yet. For the non-multi options, let's use the old selectbox.

**Comment by RyanPrice-King** [comment]
Yes, will do :)

**Comment by RyanPrice-King** [comment]
I have separated the components like you asked and noticed the select-multi was not loading pre-selected values in the dropdown, so I have fixed that now too.  Still testing and will let you know when I am happy with it.

**Review by dsternlicht — APPROVED** [review]

**Pull request merged**